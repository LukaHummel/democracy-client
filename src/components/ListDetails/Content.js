import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';

const Wrapper = styled.ScrollView`
  flex: 1;
`;

const Text = styled.Text``;

const Content = ({ description }) => (
  <Wrapper>
    <Text>{description}</Text>
  </Wrapper>
);

Content.defaultProps = {
  description: '',
};

Content.propTypes = {
  description: PropTypes.string,
};

export default Content;
