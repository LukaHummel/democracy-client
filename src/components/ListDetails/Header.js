import React from 'react';
import styled from 'styled-components/native';

import Votes from '../../components/Votes';
import Menu from '../../components/ListDetails/HeaderMenu';

const Wrapper = styled.View`
  padding-top: 27;
  padding-left: 27;
  padding-bottom: 5;
  border-bottom-width: 1;
  border-bottom-color: #c8c7cc;
`;

const HeaderMain = styled.View`
  flex-direction: row;
  justify-content: space-between;
  padding-bottom: 5;
`;

const Title = styled.Text.attrs({
  numberOfLines: 2,
})`
  flex: 1;
  font-size: 17;
  color: #030303;
  letter-spacing: -0.41;
`;

export default () => (
  <Wrapper>
    <HeaderMain>
      <Title>
        der Titelder Titelder Titelder Titelder Titelder Titelder Titelder Titelder Titelder
        Titelder Titelder Titelder Titel
      </Title>
      <Votes votes={123} />
    </HeaderMain>
    <Menu />
  </Wrapper>
);
