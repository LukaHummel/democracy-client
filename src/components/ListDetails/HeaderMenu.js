import React, { Component } from 'react';
import styled from 'styled-components/native';
import { Icon } from 'react-native-elements';

const Wrapper = styled.View`
  height: 35;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

const ItemWrapper = styled.View`
  width: 50;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`;

const MarkerIcon = styled(Icon).attrs({
  name: 'place',
})``;
const MarkerText = styled.Text``;

const CommentsIcon = styled(Icon).attrs({
  name: 'format-align-left',
})``;
const CommentText = styled.Text``;

const Date = styled.Text``;

export default class ListDetailsHeaderMenu extends Component {
  render() {
    return (
      <Wrapper>
        <ItemWrapper>
          <MarkerIcon />
          <MarkerText>234</MarkerText>
        </ItemWrapper>
        <ItemWrapper>
          <CommentsIcon />
          <CommentText>74</CommentText>
        </ItemWrapper>
        <ItemWrapper>
          <Date>3:21</Date>
        </ItemWrapper>
      </Wrapper>
    );
  }
}
