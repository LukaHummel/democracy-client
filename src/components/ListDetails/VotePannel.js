import React from 'react';
import styled from 'styled-components/native';
import Ionicons from 'react-native-vector-icons/Ionicons';
import { Alert } from 'react-native';

const Wrapper = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  height: 132;
  background-color: rgba(250, 250, 250, 0.9);
`;

const CircleButton = styled.TouchableOpacity.attrs({
  onPress: () => () => Alert.alert('du hast nun abgestimmt'),
})`
  align-items: center;
  justify-content: center;
  width: 87.5;
  height: 87.5;
  border-radius: 50;
  border-width: 1;
`;

const YesButton = styled(CircleButton)`
  border-color: #44db5e;
  background-color: rgb(68, 219, 94);
`;
const YesIcon = styled(Ionicons).attrs({
  name: 'ios-thumbs-up-outline',
  size: 50,
})``;

const NoButton = styled(CircleButton)`
  border-color: #fe3824;
  background-color: rgb(254, 56, 36);
`;
const NoIcon = styled(Ionicons).attrs({
  name: 'ios-thumbs-down-outline',
  size: 50,
})``;

const AbstentionButton = styled(CircleButton)`
  width: 63.5;
  height: 63.5;
  border-color: rgba(0, 118, 255, 0.71);
  background-color: rgba(0, 118, 255, 0.71);
`;
const AbstentionIcon = styled(Ionicons).attrs({
  name: 'ios-pause-outline',
  size: 30,
})`
  transform: rotate(90deg);
`;

export default () => (
  <Wrapper>
    <YesButton>
      <YesIcon />
    </YesButton>
    <AbstentionButton>
      <AbstentionIcon />
    </AbstentionButton>
    <NoButton>
      <NoIcon />
    </NoButton>
  </Wrapper>
);
