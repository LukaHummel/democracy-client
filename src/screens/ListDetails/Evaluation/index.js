import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';

import Pie from './../../../components/Charts/Pie';

const Wrapper = styled.View`
  flex: 1;
  width: ${props => props.width || undefined};
`;

const Evaluation = props => (
  <Wrapper width={props.width}>
    <Pie />
  </Wrapper>
);

Evaluation.propTypes = {
  width: PropTypes.number,
};

Evaluation.defaultProps = {
  width: undefined,
};

export default Evaluation;
