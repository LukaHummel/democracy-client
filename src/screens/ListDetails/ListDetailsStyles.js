import { StyleSheet } from 'react-native';

export default StyleSheet.create({
	container: {
        flex: 1,
        flexDirection: 'column'
    },
	
    header: {
        flex: .25,
        backgroundColor: '#DCDCDC'
    },
	header_box: {
        flexDirection: 'row',
    },
	title: {
		fontSize: 20,
		fontWeight: 'bold',
		flex: 1
	},
	votes: {
		width: 75,
	},
	
    body: {
        flex: .5,
        backgroundColor: '#FFF'
    },
	fulltext: {
		fontSize: 18,
	},
	
	footer: {
        flex: .25,
        backgroundColor: '#000'
    },
	buttons: {},
});