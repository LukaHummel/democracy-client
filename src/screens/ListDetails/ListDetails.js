import React, { Component } from 'react';
import styled from 'styled-components/native';

import Header from '../../components/ListDetails/Header';
import Content from '../../components/ListDetails/Content';
import VotePannel from '../../components/ListDetails/VotePannel';

const Wrapper = styled.View`
  flex: 1;
  background-color: white;
`;

export default class ListDetailsScreen extends Component {
  render() {
    return (
      <Wrapper>
        <Header />
        <Content />
        <VotePannel />
      </Wrapper>
    );
  }
}
