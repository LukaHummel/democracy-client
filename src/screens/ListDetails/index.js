import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Dimensions } from 'react-native';
import styled from 'styled-components/native';

import Header from '../../components/ListDetails/Header';
import Main from './Main';
import Evaluation from './Evaluation';

import listData from '../../data/list.json';

const Wrapper = styled.View`
  flex: 1;
  background-color: white;
`;

const Content = styled.ScrollView.attrs({
  pagingEnabled: true,
})``;

class ListDetails extends Component {
  state = {
    deviceWidth: Dimensions.get('window').width,
  };

  componentDidMount() {
    this.dimensionsListener = Dimensions.addEventListener('change', this.handleDimensionChange);
  }

  componentWillUnmount() {
    Dimensions.removeEventListener('change', this.handleDimensionChange);
  }

  handleDimensionChange = ({ window }) => {
    this.setState({ deviceWidth: window.width });
  };

  render() {
    const { id } = this.props.match.params;
    const item = listData.find(entry => entry.id === parseInt(id, 10));
    return (
      <Wrapper>
        <Header
          title={item.title}
          votes={item.votes}
          commentsCount={item.commentsCount}
          places={item.places}
        />
        <Content horizontal>
          <Main description={item.description} width={this.state.deviceWidth} />
          <Evaluation width={this.state.deviceWidth} />
        </Content>
      </Wrapper>
    );
  }
}

ListDetails.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.number.isRequired,
    }),
  }).isRequired,
};

export default ListDetails;
