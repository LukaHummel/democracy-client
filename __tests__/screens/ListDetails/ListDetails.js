import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import ListDetails from '../../../src/screens/ListDetails';

describe('test components/List/Row', () => {
  it('render row', () => {
    const props = {
      match: {
        params: {
          id: 1,
        },
      },
    };
    const tree = renderer.create(<ListDetails {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
