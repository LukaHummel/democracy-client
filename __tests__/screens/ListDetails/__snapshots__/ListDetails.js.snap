// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test components/List/Row render row 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#c8c7cc",
          "borderBottomWidth": 1,
          "paddingBottom": 5,
          "paddingLeft": 27,
          "paddingTop": 27,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 5,
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Array [
            Object {
              "color": "#030303",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontSize": 17,
              "letterSpacing": -0.41,
            },
            undefined,
          ]
        }
      >
        Gesetz zur effektiveren und praxistauglicheren Ausgestaltung des Strafverfahrens
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "width": 48,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "grey",
                "fontSize": 12,
              },
              Array [
                Object {
                  "fontSize": 45,
                  "marginBottom": -15,
                  "marginTop": -10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          adjustsFontSizeToFit={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#8f8e94",
                "fontSize": 18,
                "letterSpacing": -0.29,
              },
              undefined,
            ]
          }
        >
          632
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": 50,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#4a4a4a",
                "fontSize": 24,
              },
              Array [
                Object {},
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#4a4a4a",
                "fontSize": 12,
                "letterSpacing": -0.26,
              },
              undefined,
            ]
          }
        >
          19999
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": 50,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#4a4a4a",
                "fontSize": 24,
              },
              Array [
                Object {},
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#4a4a4a",
                "fontSize": 12,
                "letterSpacing": -0.26,
              },
              undefined,
            ]
          }
        >
          3213
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": 50,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#44db5e",
                "fontSize": 12,
                "letterSpacing": -0.26,
              },
              undefined,
            ]
          }
        >
          3:21
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    horizontal={true}
    pagingEnabled={true}
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "width": 750,
            },
            undefined,
          ]
        }
        width={750}
      >
        <RCTScrollView
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
              undefined,
            ]
          }
        >
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                ]
              }
            >
              Die staatliche Aufgabe erfordert es, die strafprozessualen Vorschriften laufend auf ihre Tauglichkeit, Zeitgemäßheit und Effektivität hin zu überprüfen und das bestehende Regelungsgefüge unter Wahrung der genannten Ziele des Strafverfahrens an die sich ändernden Rahmenbedingungen anzupassen. Der Erreichung dieses Ziels dienen die vorliegenden Gesetzgebungsvorschläge zur Effektivierung und Steigerung der Praxistauglichkeit des Strafverfahrens.
            </Text>
          </View>
        </RCTScrollView>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(250, 250, 250, 0.9)",
                "flexDirection": "row",
                "height": 132,
                "justifyContent": "space-around",
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(68, 219, 94)",
                "borderColor": "#44db5e",
                "borderRadius": 50,
                "borderWidth": 1,
                "height": 87.5,
                "justifyContent": "center",
                "opacity": 1,
                "width": 87.5,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 50,
                  },
                  Array [
                    Object {},
                    undefined,
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(0, 118, 255, 0.71)",
                "borderColor": "rgba(0, 118, 255, 0.71)",
                "borderRadius": 50,
                "borderWidth": 1,
                "height": 63.5,
                "justifyContent": "center",
                "opacity": 1,
                "width": 63.5,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "rotate": "90deg",
                        },
                      ],
                    },
                    undefined,
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(254, 56, 36)",
                "borderColor": "#fe3824",
                "borderRadius": 50,
                "borderWidth": 1,
                "height": 87.5,
                "justifyContent": "center",
                "opacity": 1,
                "width": 87.5,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 50,
                  },
                  Array [
                    Object {},
                    undefined,
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "width": 750,
            },
            undefined,
          ]
        }
        width={750}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <svg
            innerRef={[Function]}
            style={Object {}}
            viewBox="0 0 1000 1000"
          >
            <g
              style={Object {}}
              transform="translate(500, 500)"
            >
              <path
                d="M0.09902291243377474,-499.9999901944627A500,500,0,0,1,427.8752106060524,-258.6944223380672L83.39272469932911,-55.18743939722067A100,100,0,0,0,4.0999970333716425,-99.9159147700022Z"
                fill="blue"
                style={Object {}}
              />
              <path
                d="M433.0622048565918,-249.914238739507A500,500,0,0,1,499.97400019066504,-5.098934530359364L99.86991972889089,-5.098934530359187A100,100,0,0,0,88.57971894986852,-46.40725579866043Z"
                fill="green"
                style={Object {}}
              />
              <path
                d="M-499.97400019066504,-5.098934530359153A500,500,0,0,1,-10.09833625908047,-499.89801320349187L-6.097362138142701,-99.81393777903136A100,100,0,0,0,-99.86991972889089,-5.098934530359153Z"
                fill="red"
                style={Object {}}
              />
            </g>
          </svg>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
